src/Umbraco.Web.Common/UmbracoHelperAccessor.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Umbraco.Cms.Web.Common
+{
+    public class UmbracoHelperAccessor : IUmbracoHelperAccessor
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public UmbracoHelperAccessor(IHttpContextAccessor httpContextAccessor) => _httpContextAccessor = httpContextAccessor;
+
+        public bool TryGetUmbracoHelper(out UmbracoHelper umbracoHelper)
+        {
+            umbracoHelper = _httpContextAccessor.HttpContext.RequestServices.GetService<UmbracoHelper>();
```

Comment:
There is no null check here. This will fail with a `NullReferenceException` when running in a non request thread.

Reply:
I changed it to check for null 👍

Reply:
I think Shannon meant to do a null check on `_httpContextAccessor`.

Reply:
Yea i must have misunderstood, fixing that right now